autotime/__init__.py

```diff
@@ -13,26 +16,21 @@ class LineWatcher(object):
     -----
     * Register the `start` and `stop` methods with the IPython events API.
     """
-
-    def __init__(self):
-        self.start_time = 0.0
+    __slots__ = ['start_time']
 
     def start(self):
-        self.start_time = time.time()
+        self.start_time = monotonic()
 
     def stop(self):
-        stop_time = time.time()
-
-        if self.start_time:
-            diff = stop_time - self.start_time
-            assert diff >= 0
-            print('time: {}'.format(format_delta(diff)))
+        delta = monotonic() - self.start_time
+        print(u'time: {}'.format(format_delta(delta)))
 
 
 timer = LineWatcher()
 
 
 def load_ipython_extension(ip):
+    timer.start()
```

Comment:
Why does `timer.start()` need to be called here?

Reply:
so since `%load_ext autotime` will not trigger the `pre_run_cell` event (only on subsequent calls), the first time the extension is loaded, it is done manually like this.

before, with the class `__init__`, the start time would be set to 0.0, and hence a negative time would come out of the very first `post_run_cell` command (and nothing would be printed due to the if statement I removed). now, there will always be a positive delta (monotonic time, proper first call to start), and we will get a time outputted upon executing `%load_ext autotime`